src/commands/new.ts

```diff
@@ -1,137 +1,152 @@
+import { resolve } from 'node:path';
 import ora from 'ora';
-import { resolve } from 'path';
 import prompts from 'prompts';
-import { repoUrl } from '#constants';
 
-const { blueBright, red } = chalk;
+async function editPackageJson(location: string, name: string) {
+	const pjLocation = `./${location}/package.json`;
+	const output = JSON.parse(await readFile(pjLocation, 'utf8'));
+	if (!output) throw new Error("Can't read file.");
+
+	output.name = name;
 
-function editPackageJson(location: string, name: string) {
-	return new Promise(async (resolve, reject) => {
-		const pjLocation = `./${location}/package.json`;
-		const output = JSON.parse(await readFile(pjLocation, 'utf8'));
-		if (!output) return reject(new Error("Can't read file."));
+	const result = await fromAsync(async () => writeFile(pjLocation, JSON.stringify(output, null, 2)));
+
+	if (isErr(result)) {
+		return false;
+	}
 
-		output.name = name;
-		await writeFile(pjLocation, JSON.stringify(output, null, 2)).catch(reject);
-		return resolve(true);
-	});
+	return true;
 }
 
-function installDeps(location: string, pm: string, verbose: boolean) {
-	const pmp = spawn(process.platform === 'win32' ? `${pm.toLowerCase()}.cmd` : pm.toLowerCase(), ['install'], {
-		stdio: verbose ? 'inherit' : undefined,
-		cwd: `./${location}/`
-	});
-
-	return new Promise((resolve, reject) => {
-		pmp.on('error', reject);
-
-		pmp.on('exit', async (code) => {
-			if (code === 0) {
-				const lockfile = `./${location}/${pm === 'npm' ? 'yarn.lock' : 'package-lock.json'}`;
-				if (existsSync(lockfile)) await rm(lockfile);
-				resolve(true);
-			} else {
-				reject(new Error('An unknown error occured while installing the dependencies. Try running Sapphire CLI with "--verbose" flag.'));
-			}
-		});
-	});
+async function installDeps(location: string, pm: string, verbose: boolean) {
+	const result = await fromAsync(async () =>
+		execa(pm.toLowerCase(), ['install'], {
+			stdio: verbose ? 'inherit' : undefined,
+			cwd: `./${location}/`
+		})
+	);
+
+	if (isErr(result)) {
+		throw result.error;
+	}
+
+	if (result.value.exitCode !== 0) {
+		throw new Error('An unknown error occurred while installing the dependencies. Try running Sapphire CLI with "--verbose" flag.');
+	}
+
+	const oppositeLockfile = `./${location}/${pm === 'npm' ? 'yarn.lock' : 'package-lock.json'}`;
+
+	if (await fileExists(oppositeLockfile)) {
+		await rm(oppositeLockfile);
+	}
+
+	return true;
 }
 
-function configureYarnRc(location: string) {
-	return (name: string, value: string) => {
-		return new Promise((resolve, reject) => {
-			return exec(`yarn config set ${name} ${value}`, { cwd: `./${location}/` }, (e) => {
-				if (e) return reject(e);
-				return resolve(true);
-			});
-		});
-	};
+async function configureYarnRc(location: string, name: string, value: string) {
+	const result = await fromAsync(async () => execa('yarn', ['config', 'set', name, value], { cwd: `./${location}/` }));
```

Comment:
```suggestion
	const result = await fromAsync(() => execa('yarn', ['config', 'set', name, value], { cwd: `./${location}/` }));
```

Or actually, why even wrap this in fromAsync? Just await execa and return true instead